models/user/pin.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2016 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package user
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+const maxPinnedRepos = 3
+
+func GetPinnedRepositoryIDs(userID int64) ([]int64, error) {
+	pinnedstring, err := GetUserSetting(userID, PinnedRepositories)
+
+	if err != nil {
+		return nil, err
+	}
+
+	var parsedValues []int64
+	if pinnedstring == "" {
+		return parsedValues, nil
+	}
+
+	err = json.Unmarshal([]byte(pinnedstring), &parsedValues)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return parsedValues, nil
+}
+
+func setPinnedRepositories(userID int64, repos []int64) error {
+	stringed, err := json.Marshal(repos)
+
+	if err != nil {
+		return err
+	}
+
+	return SetUserSetting(userID, PinnedRepositories, string(stringed))
+
+}
+
+type TooManyPinnedReposError struct {
+	count int
+}
+
+func (e *TooManyPinnedReposError) Error() string {
+	return fmt.Sprintf("can pin at most %d repositories, %d pinned repositories is too much", maxPinnedRepos, e.count)
+}
+
+func PinRepos(ownerID int64, repoIDs ...int64) error {
+
+	repos, err := GetPinnedRepositoryIDs(ownerID)
+
+	if err != nil {
+		return err
+	}
+	newrepos := make([]int64, 0, len(repoIDs)+len(repos))
+
+	allrepos := append(repos, repoIDs...)
+
+	for _, toadd := range allrepos {
```

Comment:
Apparently, our linter complains otherwise...
```suggestion
	repos := append(repos, repoIDs...)

	for _, toadd := range repos {
```